Content fetcher database merge

tanny411 opened the pull request:
I have cleaned and updated the codes for content fetcher. I've tested by running across all wikis. Everything works fine except one out of memory issue in only one wiki that I need to scrutinize more and check. But for other wikis everything seems to work fine.

Please review the code and let me know if there are any way certain parts could be optimized or improved.

One part I think maybe optimized is to do `insert` queries for the entire dataframe at once instead of iterating through rows using `df.to_sql()`. But that seems to be possible for sqlalchemy and not pymysql. Not sure how multiple bulk queries working at once will behave though.

tanny411 committed:
Fix database queries and clean content fetcher

tanny411 committed:
Fix bugs and refactor

Retailed commented:
> I have cleaned and updated the codes for content fetcher. I've tested by running across all wikis. Everything works fine except one out of memory issue in only one wiki that I need to scrutinize more and check. But for other wikis everything seems to work fine.
> 
> Please review the code and let me know if there are any way certain parts could be optimized or improved.
> 
> One part I think maybe optimized is to do `insert` queries for the entire dataframe at once instead of iterating through rows using `df.to_sql()`. But that seems to be possible for sqlalchemy and not pymysql. Not sure how multiple bulk queries working at once will behave though.

Looks like sqalchemy [can work through pymysql](https://docs.sqlalchemy.org/en/14/dialects/mysql.html) just fine. Multiple queries like that won't damage the data and/or database, but the db could stay locked for quite a long time for huge bulk inserts, and I'm not sure if it is good in our situation.

tanny411 commented:
Does make sense.

On Fri, Dec 25, 2020, 12:55 AM lost_enchanter <notifications@github.com>
wrote:

> I have cleaned and updated the codes for content fetcher. I've tested by
> running across all wikis. Everything works fine except one out of memory
> issue in only one wiki that I need to scrutinize more and check. But for
> other wikis everything seems to work fine.
>
> Please review the code and let me know if there are any way certain parts
> could be optimized or improved.
>
> One part I think maybe optimized is to do insert queries for the entire
> dataframe at once instead of iterating through rows using df.to_sql().
> But that seems to be possible for sqlalchemy and not pymysql. Not sure how
> multiple bulk queries working at once will behave though.
>
> Looks like sqalchemy can work through pymysql
> <https://docs.sqlalchemy.org/en/14/dialects/mysql.html> just fine.
> Multiple queries like that won't damage the data and/or database, but the
> db could stay locked for quite a long time for huge bulk inserts, and I'm
> not sure if it is good in our situation.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wikimedia/abstract-wikipedia-data-science/pull/12#issuecomment-750945309>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGFZMCB6FVN5NWOO5Y6IE7TSWOFCPANCNFSM4VIOGROA>
> .
>
